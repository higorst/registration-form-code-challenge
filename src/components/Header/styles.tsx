import styled from 'styled-components/native'
import { ViewProps } from 'react-native'
import { Colors } from '../../styles/Colors'



export const TitleHeader = styled.Text`
    color: ${Colors.white};
    text-align: center;  
    font-size: 22px;
`

export const HeaderView = styled.View`
    background-color: ${Colors.headerColor};
    border-color: ${Colors.white};

    padding: 10px;
    margin-bottom: 10px;
    border-bottom-width: 1px;

    border-bottom-left-radius: 100px;
    border-bottom-right-radius: 100px;
`