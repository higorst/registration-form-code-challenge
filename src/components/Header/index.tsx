import React from 'react'
import { HeaderView, TitleHeader } from './styles'

interface PropsHeader {
    title: string
}
const Header = (props: PropsHeader) => {

    return (
        <HeaderView>  
            <TitleHeader>{props.title}</TitleHeader>          
        </HeaderView>
    )
}

export default Header