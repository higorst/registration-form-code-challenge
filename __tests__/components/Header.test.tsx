import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../src/components/Header/index'

jest.useFakeTimers()
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

describe('<Header />', () => {
    it('renders Header Component correctly', () => {
        const tree = renderer.create(
                <Header title="Snapshot"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});