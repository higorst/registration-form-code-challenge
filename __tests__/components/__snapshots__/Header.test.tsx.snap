// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders Header Component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#4e61c4",
        "borderBottomLeftRadius": 100,
        "borderBottomRightRadius": 100,
        "borderBottomWidth": 1,
        "borderColor": "#ffffff",
        "marginBottom": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "textAlign": "center",
        },
      ]
    }
  >
    Snapshot
  </Text>
</View>
`;
